tasks/flush_repo.py

```diff
@@ -84,62 +101,113 @@ async def run_async(self, db_session, *, repoid: int, **kwargs):
         db_session.commit()
+        log.info("Deleted static analysis info", extra=dict(repoid=repoid))
 
-        # label analysis
+    @sentry_sdk.trace
+    def _delete_label_analysis(self, db_session: Session, commit_ids, repoid: int):
         db_session.query(LabelAnalysisRequest).filter(
             LabelAnalysisRequest.base_commit_id.in_(commit_ids)
             | LabelAnalysisRequest.head_commit_id.in_(commit_ids)
         ).delete(synchronize_session=False)
         db_session.commit()
+        log.info("Deleted label analysis info", extra=dict(repoid=repoid))
 
-        deleted_commits = (
-            db_session.query(Commit).filter_by(repoid=repo.repoid).delete()
-        )
-        delete_branches = (
-            db_session.query(Branch).filter_by(repoid=repo.repoid).delete()
-        )
-        deleted_pulls = db_session.query(Pull).filter_by(repoid=repo.repoid).delete()
+    @sentry_sdk.trace
+    def _delete_commits(self, db_session: Session, repoid: int) -> int:
+        deleted_commits = db_session.query(Commit).filter_by(repoid=repoid).delete()
+        db_session.commit()
+        log.info("Deleted commits", extra=dict(repoid=repoid))
+        return deleted_commits
+
+    @sentry_sdk.trace
+    def _delete_branches(self, db_session: Session, repoid: int) -> int:
+        deleted_branches = db_session.query(Branch).filter_by(repoid=repoid).delete()
+        db_session.commit()
+        log.info("Deleted branches", extra=dict(repoid=repoid))
+        return deleted_branches
+
+    @sentry_sdk.trace
+    def _delete_pulls(self, db_session: Session, repoid: int) -> int:
+        deleted_pulls = db_session.query(Pull).filter_by(repoid=repoid).delete()
+        db_session.commit()
+        log.info("Deleted pulls", extra=dict(repoid=repoid))
+        return deleted_pulls
+
+    @sentry_sdk.trace
+    async def run_async(
+        self, db_session: Session, *, repoid: int, **kwargs
+    ) -> FlushRepoTaskReturnType:
+        log.info("Deleting repo contents", extra=dict(repoid=repoid))
+        repo = db_session.query(Repository).filter_by(repoid=repoid).first()
+        if repo is None:
+            log.exception("Repo not found", extra=dict(repoid=repoid))
+            return FlushRepoTaskReturnType(error="repo not found")
+
+        deleted_archives = self._delete_archive(repo)
+        with sentry_sdk.start_span("query_commit_ids"):
+            commit_ids = db_session.query(Commit.id_).filter_by(repoid=repo.repoid)
+        self._delete_comparisons(db_session, commit_ids, repoid)
+
+        with sentry_sdk.start_span("query_report_ids"):
+            report_ids = db_session.query(CommitReport.id_).filter(
+                CommitReport.commit_id.in_(commit_ids)
+            )
+        self._delete_reports(db_session, report_ids, repoid)
+        self._delete_uploads(db_session, report_ids, repoid)
+
+        self._delete_commit_details(db_session, commit_ids, repoid)
+
+        # TODO: Component comparison
```

Comment:
Would we still leave the commit and component comparisons existing in the DB? I think since commit comparisons rely on commit ids existing, that the delete_commits fn below would fail wouldnt it?

Reply:
Yes this is true. It's not the only piece of data that we should be removing but aren't (`LabelAnalysisRequestProcessingError` has the same issue).

But I wanted to be very intentional with these changes and limit them to a _refactor_.
This for a couple of reasons that include:
1. Checking all the missing pieces of data, adding them in and adding tests for all of that consumes a lot more time.
2. These changes are meant to give more information so we can make better decisions on how to improve the feature overall. We might come to the conclusion that we need to tear the whole thing apart and start from scratch. In this case the extra effort I'd have to put to add the missing details would be lost.
3. Currently there's no "epic" or "plan" on how to tackle this flow. This needs to be discussed and prioritized. I personally think that this knowledge that some data is not being deleted when it should needs to factor in on the "how big this project is gonna be?"

So while you are absolutely right and we should (and will) fix it I hope we can ignore the fact that this is partially broken for now. It is one of the reasons I left the comment there.
Also [this comment](https://github.com/codecov/engineering-team/issues/1031#issuecomment-1917575228) captures some of the data that we know is missing, in case you're worried it'll be lost.

Reply:
Yeah big time, didn't mean to imply we needed to add those as yeah this should be contained to a refactor, was mostly trying to see if what I said made sense, still learning the ropes here 😅. And ty writing the things we're missing somewhere too 👌